server/service/src/sync/test/integration/remote/patient_name_and_store_and_name_store_join.rs

```diff
@@ -0,0 +1,145 @@
+use crate::sync::{
+    test::integration::{
+        central_server_configurations::NewSiteProperties, SyncRecordTester, TestStepData,
+    },
+    translations::{IntegrationRecords, PullUpsertRecord},
+};
+use repository::{Gender, NameRow, NameStoreJoinRow, NameType, StoreRow};
+
+use serde_json::json;
+use util::{
+    inline_init,
+    uuid::{small_uuid, uuid},
+};
+
+pub(crate) struct PatientNameAndStoreAndNameStoreJoinTester;
+
+impl SyncRecordTester for PatientNameAndStoreAndNameStoreJoinTester {
+    fn test_step_data(&self, new_site_properties: &NewSiteProperties) -> Vec<TestStepData> {
+        let mut result = Vec::new();
+        // STEP 1 - insert
+        let facility_name_row = inline_init(|r: &mut NameRow| {
+            r.id = uuid();
+            r.r#type = NameType::Facility;
+            r.name = "facility".to_string();
+            r.is_customer = true;
+            r.is_supplier = true;
+        });
+        let facility_name_json = json!({
+            "ID": facility_name_row.id,
+            "type": "facility",
+            "name": "facility",
+            "customer": true,
+            "supplier": true,
+        });
+
+        let mut patient_name_row = inline_init(|r: &mut NameRow| {
+            r.id = uuid();
+            r.r#type = NameType::Patient;
+            r.is_customer = true;
+            r.is_supplier = false;
+            r.gender = Some(Gender::Male);
+        });
+        let patient_name_json = json!({
+            "ID": patient_name_row.id,
+            "type": "facility",
+            "customer": true,
+            "supplier": false,
+            "female": false,
+        });
+
+        let store_row = StoreRow {
+            id: uuid(),
+            name_id: facility_name_row.id.clone(),
+            code: small_uuid(),
+            site_id: new_site_properties.site_id as i32,
+            logo: None,
+        };
+        let store_json = json!({
+            "ID": store_row.id,
+            "code": store_row.code,
+            "name_ID": store_row.name_id,
+            "sync_id_remote_site": store_row.site_id,
+            "store_mode": "dispensary"
+        });
+
+        result.push(TestStepData {
+            central_upsert: json!({
+                "name": [patient_name_json, facility_name_json],
+                "store": [store_json]
+            }),
+            central_delete: json!({}),
+            integration_records: IntegrationRecords::from_upserts(vec![
+                PullUpsertRecord::Name(patient_name_row.clone()),
```

Comment:
patient should only sync to a store where name store join exists. and as per the docs:

https://github.com/openmsupply/open-msupply/blob/3ee43e06d8d277588c48a5310ea3fc23ebd54df4/server/service/src/sync/test/integration/README.MD#L84-L93

We would re-initialise for each test data `step` and `Check IntegrationRecords` in TestData

Reply:
[649c281](https://github.com/openmsupply/open-msupply/pull/1278/commits/649c281518cdfbd2c686e9248e6404f39c3bf302)